mitosheet/mitosheet/tests/test_parse.py

```diff
@@ -784,6 +784,132 @@ def get_string_data_for_df(columns: List[Any], length: int) -> Dict[Any, Any]:
+    (
+        '=SUM(A:B)',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 2), index=pd.RangeIndex(0, 2)),
+        'df[\'B\'] = SUM(df.loc[:, \'A\':\'B\'])',
+        set(['SUM']),
+        set(['A', "B"])
+    ),
+    # One range across two headers, with as well as other specific cell reference and header ref
+    (
+        '=SUM(A:B, A, A0)',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 2), index=pd.RangeIndex(0, 2)),
+        "df[\'B\'] = SUM(df.loc[:, \'A\':\'B\'], df['A'], df['A'])",
+        set(['SUM']),
+        set(['A', "B"])
+    ),
+]
+
+HEADER_INDEX_HEADER_INDEX_MATCHES = [
+    # Simple header index header index reference
+    (
+        '=A0:A1',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 2), index=pd.RangeIndex(0, 2)),
+        "df['B'] = RollingRange(df[['A']], 2, 0)",
+        set([]),
+        set(['A'])
+    ),
+    # Header index header index reference with offset
+    (
+        '=A1:A2',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 3), index=pd.RangeIndex(0, 3)),
+        "df['B'] = RollingRange(df[['A']], 2, -1)",
+        set([]),
+        set(['A'])
+    ),
+    # Header index header index reference with offset and two columns
+    (
+        '=A1:B2',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 3), index=pd.RangeIndex(0, 3)),
+        "df['B'] = RollingRange(df.loc[:, 'A':'B'], 2, -1)",
+        set([]),
+        set(['A', 'B'])
+    ),
+    # Header index header index reference with offset and two columns in function
+    (
+        '=SUM(A1:B2)',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 3), index=pd.RangeIndex(0, 3)),
+        "df['B'] = SUM(RollingRange(df.loc[:, 'A':'B'], 2, -1))",
+        set(['SUM']),
+        set(['A', 'B'])
+    ),
+    # Header index header index reference with offset and two columns in function, as well as other specific cell reference and header ref
+    (
+        '=SUM(A1:B2, A, A0)',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 3), index=pd.RangeIndex(0, 3)),
+        "df['B'] = SUM(RollingRange(df.loc[:, 'A':'B'], 2, -1), df['A'], df['A'])",
+        set(['SUM']),
+        set(['A', 'B'])
+    ),
+    # Header index header index as well as header header
+    (
+        '=SUM(A1:B2, A:B)',
+        'B',
+        0,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 3), index=pd.RangeIndex(0, 3)),
+        "df['B'] = SUM(RollingRange(df.loc[:, 'A':'B'], 2, -1), df.loc[:, 'A':'B'])",
+        set(['SUM']),
+        set(['A', 'B'])
+    ),
+    # Header index header index reference with offset
+    (
+        '=A3:A7',
+        'B',
+        5,
+        pd.DataFrame(get_number_data_for_df(['A', 'B'], 10), index=pd.RangeIndex(0, 10)),
+        "df['B'] = RollingRange(df[['A']], 5, -2)",
```

Comment:
I added this test and it is failing. We expect this to be `-2` offset, but the parser is returning `2`.